pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -110,6 +110,16 @@ type ActualStateOfWorld interface {
 	// volumes that do not need to update contents should not fail.
 	PodExistsInVolume(podName volumetypes.UniquePodName, volumeName v1.UniqueVolumeName) (bool, string, error)
 
+	// PodRemovedFromVolume returns true if the given pod does not in the list of
+	// mountedPods for the given volume in the cache, indicating that the pod has
+	// fully unmounted it or it was never mounted.
+	// If a pod with the same unique name does not exist under the specified
+	// volume, true is returned.
+	// If a volume with the name volumeName does not exist in the list of
+	// attached volumes, true is returned indicating the given volume is not yet
+	// attached.
+	PodRemovedFromVolume(podName volumetypes.UniquePodName, volumeName v1.UniqueVolumeName) bool
```

Comment:
typo in - "given pod does not exist"?

I would perhaps also add the context that - 

> If a volume is fully mounted or is in uncertain mount state for a pod, it is considered that pod still exists in volume manager's actual state of the world.

Reply:
Rephrased a bit, using your suggestion.

```
	// PodRemovedFromVolume returns true if the given pod does not exist in the list of
	// mountedPods for the given volume in the cache, indicating that the pod has
	// fully unmounted it or it was never mounted the volume.
	// If the volume is fully mounted or is in uncertain mount state for the pod, it is
	// considered that the pod still exists in volume manager's actual state of the world
	// and false is returned.
```

Reply:
how about name the function PodVolumeNotExist?

Reply:
I think it would be confusing with `PodExistsInVolume` - these two methods are not simple negation of each other. I intentionally chose a different name.